components/prism-mongodb.js

```diff
@@ -0,0 +1,100 @@
+		'$match', '$merge', '$out', '$planCacheStats', '$project', '$redact', '$replaceRoot', '$replaceWith', 
+		'$sample', '$set', '$skip', '$sort', '$sortByCount', '$unionWith', '$unset', '$unwind',
+
+		// aggregation pipeline operators
+		'$abs', '$accumulator', '$acos', '$acosh', '$add', '$addToSet', '$allElementsTrue', '$and', 
+		'$anyElementTrue', '$arrayElemAt', '$arrayToObject', '$asin', '$asinh', '$atan', '$atan2',
+		'$atanh', '$avg', '$binarySize', '$bsonSize', '$ceil', '$cmp', '$concat', '$concatArrays', '$cond', 
+		'$convert', '$cos', '$dateFromParts', '$dateToParts', '$dateFromString', '$dateToString', '$dayOfMonth', 
+		'$dayOfWeek', '$dayOfYear', '$degreesToRadians', '$divide', '$eq', '$exp', '$filter', '$first',
+		'$floor', '$function', '$gt', '$gte', '$hour', '$ifNull', '$in', '$indexOfArray', '$indexOfBytes', 
+		'$indexOfCP', '$isArray', '$isNumber', '$isoDayOfWeek', '$isoWeek', '$isoWeekYear', '$last',
+		'$last', '$let', '$literal', '$ln', '$log', '$log10', '$lt', '$lte', '$ltrim', '$map', '$max',
+		'$mergeObjects', '$meta', '$min', '$millisecond', '$minute', '$mod', '$month', '$multiply', '$ne', 
+		'$not', '$objectToArray', '$or', '$pow', '$push', '$radiansToDegrees', '$range', '$reduce',
+		'$regexFind', '$regexFindAll', '$regexMatch', '$replaceOne', '$replaceAll', '$reverseArray', '$round', 
+		'$rtrim', '$second', '$setDifference', '$setEquals', '$setIntersection', '$setIsSubset', '$setUnion', 
+		'$size', '$sin', '$slice', '$split', '$sqrt', '$stdDevPop', '$stdDevSamp', '$strcasecmp', '$strLenBytes',
+		'$strLenCP', '$substr', '$substrBytes', '$substrCP', '$subtract', '$sum', '$switch', '$tan',
+		'$toBool', '$toDate', '$toDecimal', '$toDouble', '$toInt', '$toLong', '$toObjectId', '$toString',
+		'$toLower', '$toUpper', '$trim', '$trunc', '$type', '$week', '$year', '$zip',
+
+		// aggregation pipeline query modifiers
+		'$comment', '$explain', '$hint', '$max', '$maxTimeMS', '$min', '$orderby', '$query',
+		'$returnKey', '$showDiskLoc', '$natural',
+	];
+
+	var builtinFunctions = [
+		'ObjectId',
+		'Code',
+		'BinData',
+		'DBRef',
+		'Timestamp',
+		'NumberLong',
+		'NumberDecimal',
+		'MaxKey',
+		'MinKey',
+		'RegExp',
+		'ISODate',
+		'UUID',
+	];
+
+	operators = operators.map(function(operator) { 
+		return operator.replace('$', '\\$');
+	});
+
+	var operatorsRegex = '(?:' + operators.join('(?:\\b|:)|') + ')\\b';
+
+	Prism.languages.mongodb = Prism.languages.extend('javascript', {});
+
+	Prism.languages.insertBefore('mongodb', 'string', {
+		'property': {
+			pattern: /(?:(["'])(?:\\(?:\r\n|[\s\S])|(?!\1)[^\\\r\n])*\1|[_$a-zA-Z\xA0-\uFFFF][$\w\xA0-\uFFFF]*)(?=\s*:)/,
+			greedy: true,
+			inside: {
+				'keyword': RegExp('^([\'"])?' + operatorsRegex + '(?:\\1)?$')
+			}
+		}
+	});
+
+	Prism.languages.mongodb.string.inside = {
+		url: {
+			// url pattern
+			pattern: /https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b[-a-zA-Z0-9()@:%_\+.~#?&//=]*/,  
```

Comment:
Same as before. I think URLs and IP addresses should be highlighted only if the whole string is one to prevent false positives.

Also, the pattern can be simplified.

```suggestion
			pattern: /https?:\/\/[-\w@:%.+~#=]{1,256}\.[a-z0-9()]{1,6}\b[-\w()@:%+.~#?&/=]*/i,  
```

Reply:
Resolved [here](https://github.com/PrismJS/prism/pull/2518#issuecomment-673480969).